addons/project/static/src/components/subtask_kanban_list/subtask_kanban_list.js

```diff
@@ -38,44 +43,45 @@ export class SubtaskKanbanList extends Component {
         });
     }
 
-    get fieldsInfo() {
+    get fieldNames() {
+        return Object.keys(this.fields);
+    }
+
+    get fields() {
+        const { display_name, state, user_ids } = this.props.record.fields;
         return {
-            child_ids: {
-                type: "one2many",
-                relation: "project.task",
-                relatedFields: {
-                    display_name: { type: "char" },
-                    state: {
-                        selection: [
-                            ["01_in_progress", "In Progress"],
-                            ["02_changes_requested", "Changes Requested"],
-                            ["03_approved", "Approved"],
-                            ["04_waiting_normal", "Waiting"],
-                            ["1_done", "Done"],
-                            ["1_canceled", "Canceled"],
-                        ],
-                        string: "Status",
-                        type: "selection",
-                        field: this.props.record.activeFields.state.field,
-                        viewType: this.props.record.activeFields.state.viewType,
-                        attrs: this.props.record.activeFields.state.attrs,
-                        options: this.props.record.activeFields.state.options,
-                    },
-                    user_ids: {
-                        type: "many2many",
-                        relation: "res.users",
-                        field: this.props.record.activeFields.user_ids.field,
-                        relatedFields: this.props.record.activeFields.user_ids.relatedFields,
-                        attrs: this.props.record.activeFields.user_ids.attrs,
-                        options: this.props.record.activeFields.user_ids.options,
-                    },
-                },
+            display_name,
+            state,
+            user_ids,
+        }
+    }
+
+    get activeFields() {
+        return {
+            display_name: {},
+            state: {
+                viewType: "kanban",
+                field: getFieldFromRegistry(this.fields.state.type, "project_task_state_selection", "kanban"),
+            },
+            user_ids: {
+                field: getFieldFromRegistry(this.fields.user_ids.type, "many2many_avatar_user", "kanban"),
             },
-        };
+        }
     }
 
-    get values() {
-        return { child_ids: this.subTaskIds };
+    async onSubTaskSaved(subTask) {
+        let toReload = false;
+        if (this.subTaskClosed.has(subTask.resId) && !subTask.data.state.startsWith("1_")) {
+            this.subTaskClosed.delete(subTask.resId);
+            toReload = true;
+        } else if (!this.subTaskClosed.has(subTask.resId) && subTask.data.state.startsWith("1_")) {
+            this.subTaskClosed.add(subTask.resId);
+            toReload = true;
+        }
+        if (toReload) {
+            await this.props.record.load();
+            this.props.record.model.notify();
+        }
```

Comment:
```suggestion
        const isKnownAsClosed = this.subTaskClosed.has(subTask.resId);
        const isClosed = subTask.data.state.startsWith("1_");
        if (isKnownAsClosed && !isClosed) {
            this.subTaskClosed.delete(subTask.resId);
            toReload = true;
        } else if (!isKnownAsClosed && isClosed) {
            this.subTaskClosed.add(subTask.resId);
            toReload = true;
        } else return;
        await this.props.record.load();
        this.props.record.model.notify();
```
Here's an idea :shrug:

Reply:
Good idea, thanks :)